src/hotspot/cpu/riscv/riscv_v.ad

```diff
@@ -3321,6 +3394,7 @@ instruct vcvtLtoI(vReg dst, vReg src) %{
 instruct vcvtLtoF(vReg dst, vReg src) %{
   predicate(Matcher::vector_element_basic_type(n) == T_FLOAT);
   match(Set dst (VectorCastL2X src));
+  effect(TEMP_DEF dst);
```

Comment:
I see you added `TEMP_DEF dst` for some existing instructs like this one here. Do we really need it?
I don't see such a need when reading the overlap constraints on vector operands from the RVV spec [1]:

```
A destination vector register group can overlap a source vector register group only if one of the following holds:

    The destination EEW equals the source EEW.

    The destination EEW is smaller than the source EEW and the overlap is in the lowest-numbered part of the source register group (e.g., when LMUL=1, vnsrl.wi v0, v0, 3 is legal, but a destination of v1 is not).

    The destination EEW is greater than the source EEW, the source EMUL is at least 1, and the overlap is in the highest-numbered part of the destination register group (e.g., when LMUL=8, vzext.vf4 v0, v6 is legal, but a source of v0, v2, or v4 is not).
```

[1] https://github.com/riscv/riscv-v-spec/blob/master/v-spec.adoc#sec-vec-operands

Reply:
You're right, thanks for sharing the information.